client/my-sites/plans-grid/components/plan-type-selector/hooks/use-interval-options.tsx

```diff
@@ -0,0 +1,102 @@
+import { UrlFriendlyTermType } from '@automattic/calypso-products';
+import { LocalizeProps, TranslateResult, useTranslate } from 'i18n-calypso';
+import { IntervalTypeProps, SupportedUrlFriendlyTermType } from '../types';
+import generatePath from '../utils';
+import useTermViseMaxDiscounts from './use-term-vise-max-dicounts';
+
+const getDiscountText = ( discountPercentage: number, translate: LocalizeProps[ 'translate' ] ) => {
+	if ( ! discountPercentage ) {
+		return '';
+	}
+	return translate( 'upto %(discount)d% off', {
+		args: { discount: discountPercentage },
+		comment: 'Discount percentage',
+	} );
+};
+export default function useIntervalOptions( props: IntervalTypeProps ): Record<
+	SupportedUrlFriendlyTermType,
+	{
+		key: string;
+		name: TranslateResult;
+		discountText: TranslateResult;
+		url: string;
+		termInMonths: number;
+		ui?: any;
+	}
+> {
+	const translate = useTranslate();
+	const optionList: Record<
+		SupportedUrlFriendlyTermType,
+		{
+			key: string;
+			name: TranslateResult;
+			discountText: TranslateResult;
+			url: string;
+			termInMonths: number;
+		}
+	> = {
+		yearly: {
+			key: 'yearly',
+			name: translate( 'Pay yearly' ),
+			discountText: '',
+			url: '',
+			termInMonths: 12,
+		},
+		'2yearly': {
+			key: '2yearly',
+			name: translate( 'Pay every 2 years' ),
+			discountText: '',
+			url: '',
+			termInMonths: 24,
+		},
+		'3yearly': {
+			key: '3yearly',
+			name: translate( 'Pay every 3 years' ),
+			discountText: '',
+			url: '',
+			termInMonths: 36,
+		},
+		monthly: {
+			key: 'monthly',
+			name: translate( 'Pay monthly' ),
+			discountText: '',
+			url: '',
+			termInMonths: 1,
+		},
+	};
+	const termViseDiscounts = useTermViseMaxDiscounts(
+		props.plans,
+		Object.keys( optionList ) as Array< UrlFriendlyTermType >,
+		props.usePricingMetaForGridPlans
+	);
+
+	const additionalPathProps = {
+		...( props.redirectTo ? { redirect_to: props.redirectTo } : {} ),
+		...( props.selectedPlan ? { plan: props.selectedPlan } : {} ),
+		...( props.selectedFeature ? { feature: props.selectedFeature } : {} ),
+	};
+
+	const isDomainUpsellFlow = new URLSearchParams( window.location.search ).get( 'domain' );
+
+	const isDomainAndPlanPackageFlow = new URLSearchParams( window.location.search ).get(
+		'domainAndPlanPackage'
+	);
+
+	const isJetpackAppFlow = new URLSearchParams( window.location.search ).get( 'jetpackAppPlans' );
+
+	Object.keys( optionList ).forEach( ( key ) => {
+		optionList[ key as SupportedUrlFriendlyTermType ] = {
```

Comment:
`Object.fromEntries` seems to be the best practice nowadays for such transformations. So in this case, something like

```
Object.fromEntries(
  Object.keys( optionsList ).map( key => [ key, { ... } ] )
)
```

Reply:
it's a tiny bit more performant than a `reduce` call with spreading, but certainly not a match performance-wise to mutating am object. I generally avoid mutations like fire, so up to taste

It's semantically more geared towards explaining the operation happening e.g. if you have an array of entries and you want to extract an object from it (then "fromEntries" captures that well). I think we'll be seeing a lot more of it going forward, so probably good to get into a habbit.